dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/handling/DataHandler.java

```diff
@@ -0,0 +1,1161 @@
+        checkNotNull( queryValidator );
+        checkNotNull( systemSettingManager );
+        checkNotNull( analyticsManager );
+        checkNotNull( organisationUnitService );
+
+        this.eventAnalyticsService = eventAnalyticsService;
+        this.rawAnalyticsManager = rawAnalyticsManager;
+        this.constantService = constantService;
+        this.resolver = resolver;
+        this.expressionService = expressionService;
+        this.queryPlanner = queryPlanner;
+        this.queryValidator = queryValidator;
+        this.systemSettingManager = systemSettingManager;
+        this.analyticsManager = analyticsManager;
+        this.organisationUnitService = organisationUnitService;
+    }
+
+    /**
+     * Adds indicator values to the given grid based on the given data query
+     * parameters.
+     *
+     * @param params the {@link DataQueryParams}.
+     * @param grid the grid.
+     */
+    void addIndicatorValues( DataQueryParams params, Grid grid )
+    {
+        if ( !params.getIndicators().isEmpty() && !params.isSkipData() )
+        {
+            DataQueryParams dataSourceParams = newBuilder( params )
+                .retainDataDimension( INDICATOR )
+                .withIncludeNumDen( false ).build();
+
+            List<Indicator> indicators = asTypedList( dataSourceParams.getIndicators() );
+
+            List<Period> filterPeriods = dataSourceParams.getTypedFilterPeriods();
+
+            Map<String, Constant> constantMap = constantService.getConstantMap();
+
+            // -----------------------------------------------------------------
+            // Get indicator values
+            // -----------------------------------------------------------------
+
+            Map<String, Map<String, Integer>> permutationOrgUnitTargetMap = getOrgUnitTargetMap( dataSourceParams,
+                indicators );
+
+            List<List<DimensionItem>> dimensionItemPermutations = dataSourceParams.getDimensionItemPermutations();
+
+            Map<String, List<DimensionItemObjectValue>> permutationDimensionItemValueMap = getPermutationDimensionItemValueMap(
+                dataSourceParams );
+
+            handleEmptyDimensionItemPermutations( dimensionItemPermutations );
+
+            for ( Indicator indicator : indicators )
+            {
+                for ( List<DimensionItem> dimensionItems : dimensionItemPermutations )
+                {
+                    IndicatorValue value = getIndicatorValue( filterPeriods, constantMap, permutationOrgUnitTargetMap,
+                        permutationDimensionItemValueMap, indicator, dimensionItems );
+
+                    addIndicatorValuesToGrid( params, grid, dataSourceParams, indicator, dimensionItems, value );
+                }
+            }
+        }
+    }
+
+    /**
+     * Based on the given Indicator plus additional parameters, this method will
+     * find the respective IndicatorValue.
+     * 
+     * @param filterPeriods
```

Comment:
Remove `@param`(s) or add description for the arguments

Reply:
Done